src/app/features/tech-record/components/tech-record-generate-letter/tech-record-generate-letter.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, OnChanges, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormGroup, FormNode, FormNodeOption, FormNodeTypes, FormNodeWidth } from '@forms/services/dynamic-form.types';
+import { TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { Store } from '@ngrx/store';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { generateLetter, generateLetterSuccess, generatePlate, generatePlateSuccess } from '@store/technical-records';
+import { TechnicalRecordServiceState } from '@store/technical-records/reducers/technical-record-service.reducer';
+import { take } from 'rxjs';
+import { ValidatorNames } from '@forms/models/validators.enum';
+import { Actions, ofType } from '@ngrx/effects';
+import { Letters } from '@api/vehicle/model/letters';
+
+@Component({
+  selector: 'app-generate-letter',
+  templateUrl: './tech-record-generate-letter.component.html',
+  styleUrls: ['./tech-record-generate-letter.component.scss']
+})
+export class GenerateLetterComponent implements OnInit, OnChanges {
+  reasons: Array<FormNodeOption<string>> = [
```

Comment:
Because the `reasons` is never mutated and should never be mutated, it would be ideal for this to be a getter.